extension/src/process/execution.ts

```diff
@@ -2,11 +2,36 @@ import { ChildProcess } from 'child_process'
 import { Readable } from 'stream'
 import { Event, EventEmitter } from 'vscode'
 import { Disposable } from '@hediet/std/disposable'
-import execa from 'execa'
-import doesProcessExists from 'process-exists'
+import { Deferred } from '@hediet/std/synchronization'
 import kill from 'tree-kill'
 import { getProcessPlatform } from '../env'
 
+const deferred = new Deferred()
+export const esmModulesImported = deferred.promise
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+type EsmExeca = typeof import('execa').execa
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+type EsmProcessExists = typeof import('process-exists').processExists
+
+const envCanImportEsm = process.env.NODE_ENV !== 'test'
+
+let execa: EsmExeca
+let doesProcessExist: EsmProcessExists
+const importEsmModules = async () => {
```

Comment:
[F] The reason that we cannot simply `await import('execa')` before we call it is that whenever we create a process the function has to be sync. If a `Promise<Process>` is returned then we cannot attach anything to it. We can only `await` the `stdout`.

Reply:
this is still relevant